### PR TITLE
fix: add json-ld script handling

### LINKED ISSUE #533: add support for JSON-LD in script tags
Make the html2md conversion symmetrical with the pipelines md2html by adding support for JSON-LD in `script` tags within `head`. Similar to how [meta tags are currently handled](https://github.com/adobe/helix-html2md/blob/main/src/html2md.js#L86), insert the value into the metadata table.

### PR DESCRIPTION
**Pull request by maxakuru**
fix #533

### CODE REVIEW

File: src/html2md.js [hunk 1]
```diff
@@ -92,6 +97,12 @@ function addMetadata(hast, mdast) {
           meta.set(text(name), text(content));
         }
       }
+    } else if (child.tagName === 'script' && child.properties.type === 'application/ld+json') {
+      try {
+        meta.set(text('json-ld'), text(sanitizeJsonLd(toString(child))));
```
Comment (tripodsan) [inline src/html2md.js:102]: why sanitize? those should be escaped by markdown to string

File: test/html2md.test.js [hunk 2]
```diff
@@ -19,10 +19,10 @@ import { Nock } from './utils.js';
 
 async function test(spec) {
   const html = await readFile(resolve(__testdir, 'fixtures', `${spec}.html`), 'utf-8');
-  const expected = await readFile(resolve(__testdir, 'fixtures', `${spec}.md`), 'utf-8');
   const actual = await html2md(html, {
     log: console,
   });
+  const expected = await readFile(resolve(__testdir, 'fixtures', `${spec}.md`), 'utf-8');
```
Comment (tripodsan) [inline test/html2md.test.js:25]: why did you move this?

File: test/html2md.test.js [hunk 3]
```diff
@@ -95,6 +95,18 @@ describe('html2md Tests', () => {
   it('convert a document with self-closing breaks correctly', async () => {
     await test('self-closing-breaks');
   });
+
+  it('convert a document with self-closing breaks correctly', async () => {
+    await test('self-closing-breaks');
+  });
```
Comment (tripodsan) [inline test/html2md.test.js:101]: duplicate code

**Review by tripodsan — CHANGES_REQUESTED** [review]
I don't think we need to replace the `<` in the ld.
also, is it really a good idea to put potential megabytes of json-ld in the per-document metadata? maybe we can add a limt... maybe 2k ?

File: test/html2md.test.js (hunk 2 above)
Comment (maxakuru) [reply test/html2md.test.js:25]: one of the added tests doesn't have an actual to compare, it should throw on the conversion

File: src/html2md.js (hunk 1 above)
Comment (maxakuru) [reply src/html2md.js:102]: ok, I think parsing to ensure the json is valid is useful at least

**Comment by dylandepass** [comment]
> maybe we can add a limt... maybe 2k ?

I think 2k is going to be too restrictive. I just checked an example with one of our customers and the JSON LD for a Product was 14k.

**Comment by maxakuru** [comment]
AFAICS we also have no metadata size limits anywhere else for html2md? maybe it's enough that we limit the entire html string size to 1mb? https://github.com/adobe/helix-html2md/blob/main/src/index.js#L145

**Review by tripodsan — APPROVED** [review]

**Pull request merged**